python/legate_dataframe/ldf_polars/dsl/expressions/binaryop.py

```diff
@@ -89,6 +89,17 @@ def do_evaluate(
     ) -> Column:
         """Evaluate this expression given a dataframe for context."""
         left, right = (child.evaluate(df, context=context) for child in self.children)
-        lop = left.obj
-        rop = right.obj
-        return Column(binary_operation(lop, rop, self.op, self.dtype))
+
+        if self.op != "true_divide":
+            return Column(binary_operation(left.obj, right.obj, self.op, self.dtype))
+
+        # Use divide, but cast one of the inputs to the output dtype
+        # to ensure the right type is used (with some logic to prefer
+        # casting a scalar).
+        if left.obj.type() != self.dtype and right.obj.type() != self.dtype:
+            if left.is_scalar:
+                left = left.astype(self.dtype)
+            else:
+                right = right.astype(self.dtype)
```

Comment:
I figured that the result dtype should be a correct float and casting the input is fine and probably what would happen (internally thought).
But thought to skip if clearly not needed and to prefer casting the scalar side if we have one.